### PR TITLE
Ability to define stacksize when creating a new thread

### PR DESCRIPTION
**Pull request by aldrik-ramaekers**

### CODE REVIEW

**Comment by SiegeLord** [comment]
This looks mostly good, thanks. One thing that needs to be fixed is formatting: in Allegro we use 3 space indentation, while you have a mix of spaces and a few tabs. Would you mind fixing that? After that, I think it's good to go.

File: docs/src/refman/threads.txt [hunk 1]
```diff
@@ -45,6 +45,19 @@ See also: [al_start_thread], [al_join_thread].
 
 
 
+
+## API: al_create_thread_with_stacksize
```
[the remark this answers is not in the document]
Comment (SiegeLord) [reply docs/src/refman/threads.txt]: Oh yeah, about that. So I definitely had the same reaction as you initially, but then looking at the implementation of the Win32 side of things this is literally the only argument that API supports. In that vein it is similar to things like `al_load_ttf_font_stretch`.

The alternatives I had come with were:
- al_set_new_thread_stack_size(int stack_size)
- al_create_thread_with_options(..., ALLEGRO_THREAD_OPTIONS*), with the options struct with a bunch of reserved fields for future expansion.

**Comment by dos1** [comment]
This breaks the build with SDL2 backend. I won't even mind if it's just a dummy function there, but it should at least compile and link properly :P

**Comment by SiegeLord** [comment]
We should start writing these things using ALLEGRO_SYSTEM_INTERFACE (and convert existing ones as well), so that we don't need to implement it for every single system simultaneously...

File: docs/src/refman/threads.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (SiegeLord) [reply docs/src/refman/threads.txt]: Hah, and looks like SDL defines this exact function too. Well, let's keep it. As with most new API it'll be unstable and we can refine it as necessary.

**Comment by SiegeLord** [comment]
This looks great, @aldrikboy, thanks for doing a proper implementation for SDL backend. Would you mind renaming the internal `_al_thread_with_stacksize_create` to `_al_thread_create_with_stacksize`? Other than that it looks great.

File: docs/src/refman/threads.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedro-w) [reply docs/src/refman/threads.txt]: I had the [same reaction](https://www.allegro.cc/forums/thread/617720/1041110#target) too, but I think Siegelord's analysis is right.

**Pull request merged**